test/e2e/case18_compliance_api_test.go

```diff
@@ -112,4 +110,886 @@ var _ = Describe("Test policy webhook", Label("compliance-events-api"), Ordered,
+			payload1 := []byte(`{
+				"cluster": {
+					"name": "cluster4",
+					"cluster_id": "test3-cluster4-fake-uuid-4"
+				},
+				"parent_policy": {
+					"name": "policies.common-parent",
+					"categories": ["cat-3", "cat-4"],
+					"controls": ["ctrl-2"],
+					"standards": ["stand-2"]
+				},
+				"policy": {
+					"apiGroup": "policy.open-cluster-management.io",
+					"kind": "ConfigurationPolicy",
+					"name": "common",
+					"spec": "{\"test\":\"three\",\"severity\":\"low\"}"
+				},
+				"event": {
+					"compliance": "NonCompliant",
+					"message": "configmaps [common] not found in namespace default",
+					"timestamp": "2023-03-03T03:03:03.333Z"
+				}
+			}`)
+
+			// payload2 just uses the specHash for the policy.
+			// Note that severity needs to be specified
+			payload2 := []byte(`{
+				"cluster": {
+					"name": "cluster4",
+					"cluster_id": "test3-cluster4-fake-uuid-4"
+				},
+				"parent_policy": {
+					"name": "policies.common-parent",
+					"categories": ["cat-3", "cat-4"],
+					"controls": ["ctrl-2"],
+					"standards": ["stand-2"]
+				},
+				"policy": {
+					"apiGroup": "policy.open-cluster-management.io",
+					"kind": "ConfigurationPolicy",
+					"name": "common",
+					"specHash": "5382228c69c6017d4efbd6e42717930cb2020da0",
+					"severity": "low"
+				},
+				"event": {
+					"compliance": "NonCompliant",
+					"message": "configmaps [common] not found in namespace default",
+					"timestamp": "2023-04-04T04:04:04.444Z"
+				}
+			}`)
+
+			BeforeAll(func(ctx context.Context) {
+				By("POST the events")
+				Eventually(postEvent(ctx, payload1), "5s", "1s").ShouldNot(HaveOccurred())
+				Eventually(postEvent(ctx, payload2), "5s", "1s").ShouldNot(HaveOccurred())
+			})
+
+			It("Should have only created one cluster in the table", func() {
+				rows, err := db.Query("SELECT * FROM clusters WHERE name = $1", "cluster4")
+				Expect(err).ToNot(HaveOccurred())
+
+				count := 0
+				for rows.Next() {
+					var (
+						id        int
+						name      string
+						clusterID string
+					)
+					err := rows.Scan(&id, &name, &clusterID)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(id).NotTo(Equal(0))
+					count++
+				}
+
+				Expect(count).To(Equal(1))
+			})
+
+			It("Should have only created one parent policy in a table", func() {
+				rows, err := db.Query("SELECT * FROM parent_policies WHERE name = $1", "policies.common-parent")
+				Expect(err).ToNot(HaveOccurred())
+
+				count := 0
+				for rows.Next() {
+					var (
+						id     int
+						name   string
+						cats   pq.StringArray
+						ctrls  pq.StringArray
+						stands pq.StringArray
+					)
+
+					err := rows.Scan(&id, &name, &cats, &ctrls, &stands)
+					Expect(err).ToNot(HaveOccurred())
+
+					Expect(id).NotTo(Equal(0))
+					count++
+				}
+
+				Expect(count).To(Equal(1))
+			})
+
+			It("Should have only created one policy in a table", func() {
```

Comment:
This should be 2 policies right? Since severity is set on one but not the other?

Reply:
In this test, payload1 has the severity inside the spec. Since payload2 uses specHash, the severity can't be extracted in the same way, and specifying the severity separately is necessary.

Realistically, the caller should probably always specify severity...

Reply:
Oh I see.... I'm thinking that we shouldn't try to parse the severity in this API. What do you think?

I could go either way but I lean towards not trying to parse it.

Reply:
Obviously it's already coded to parse the severity, so I thought it was a good idea 😆 

I suppose requiring the severity separately in some situations, but not others, could be annoying to the user.

Reply:
Yeah, I think reducing the "magic" is better in this instance.  :smile: